### DISCUSSION TITLE
Ignore "only" and "skip" in run mode

### DISCUSSION THREAD

**Discussion by unlocomqx** [cypress-io/cypress#29707] [Questions and Help]
Hi
Is there a way to ignore .only and .skip in run mode?
Is there a setting or a command line flag that I can use?
Thank you

**Comment by portyu9** [comment]
Hi @unlocomqx 

There is no Cypress `run` flag that means:

> ignore every `.only` and `.skip` and execute all tests anyway.

Those modifiers are part of the Mocha test definitions Cypress uses:

```js
it.only(...)
it.skip(...)
describe.only(...)
describe.skip(...)
```

By the time Cypress executes the spec, they have already changed which tests Mocha considers runnable.

For accidental committed `.only`, I would prevent it in CI with ESLint rather than trying to reinterpret it at runtime.

For example, with `eslint-plugin-mocha`:

```js
{
  rules: {
    'mocha/no-exclusive-tests': 'error',
  },
}
```

Then make linting part of CI:

```sh
npm run lint
npx cypress run
```

You can similarly reject committed `.skip` calls with:

```js
{
  rules: {
    'mocha/no-pending-tests': 'error',
  },
}
```

if your project policy is that CI must not contain skipped tests.

This is also the approach used by Cypress's own repository for `.only`: exclusive tests are rejected by linting.

If `.skip` is being used intentionally but you sometimes want those tests enabled in CI, I would replace it with an explicit environment/tag condition rather than trying to override Mocha globally. For example:

```js
const runOptional = Cypress.env('RUN_OPTIONAL')

const optionalIt = runOptional ? it : it.skip

optionalIt('optional scenario', () => {
  // ...
})
```

Then the decision to include those tests is explicit.

So the short answer is: **no runtime “ignore only/skip” mode; enforce `.only` through linting and make conditional skips explicit if you need different CI behavior.**

Please mark this answer as accepted if it helped, thank you.
